vnext/Chakra/ChakraJsiRuntime.h

```diff
@@ -0,0 +1,241 @@
+    jsi::Object&,
+    const jsi::PropNameID& name,
+    const jsi::Value& value) override;
+  bool isArray(const jsi::Object&) const override;
+  bool isArrayBuffer(const jsi::Object&) const override;
+  bool isFunction(const jsi::Object&) const override;
+  bool isHostObject(const jsi::Object&) const override;
+  bool isHostFunction(const jsi::Function&) const override;
+  jsi::Array getPropertyNames(const jsi::Object&) override;
+
+  jsi::WeakObject createWeakObject(const jsi::Object&) override;
+  jsi::Value lockWeakObject(const jsi::WeakObject&) override;
+
+  jsi::Array createArray(size_t length) override;
+  size_t size(const jsi::Array&) override;
+  size_t size(const jsi::ArrayBuffer&) override;
+  uint8_t* data(const jsi::ArrayBuffer&) override;
+  jsi::Value getValueAtIndex(const jsi::Array&, size_t i) override;
+  void setValueAtIndexImpl(jsi::Array&, size_t i, const jsi::Value& value) override;
+
+  jsi::Function createFunctionFromHostFunction(
+    const jsi::PropNameID& name,
+    unsigned int paramCount,
+    jsi::HostFunctionType func) override;
+  jsi::Value call(
+    const jsi::Function&,
+    const jsi::Value& jsThis,
+    const jsi::Value* args,
+    size_t count) override;
+  jsi::Value callAsConstructor(
+    const jsi::Function&,
+    const jsi::Value* args,
+    size_t count) override;
+
+  bool strictEquals(const jsi::String& a, const jsi::String& b) const override;
+  bool strictEquals(const jsi::Object& a, const jsi::Object& b) const override;
+  bool instanceOf(const jsi::Object& o, const jsi::Function& f) override;
+
+private:
+  // Basically convenience casts
+  static JsValueRef stringRef(const jsi::String& str);
+  static JsPropertyIdRef propIdRef(const jsi::PropNameID& sym);
+  static JsValueRef objectRef(const jsi::Object& obj);
+
+  // Factory methods for creating String/Object
+  jsi::String createString(JsValueRef stringRef) const;
+  jsi::PropNameID createPropNameID(JsValueRef stringRef);
+  jsi::Object createObject(JsValueRef objectRef) const;
+
+  // Used by factory methods and clone methods
+  jsi::Runtime::PointerValue* makeStringValue(JsValueRef str) const;
+  jsi::Runtime::PointerValue* makeObjectValue(JsValueRef obj) const;
+  jsi::Runtime::PointerValue* makePropertyIdValue(JsPropertyIdRef propId) const;
+
+  inline void checkException(JsErrorCode res);
+  inline void checkException(JsErrorCode res, const char* msg);
+
+  JsRuntimeHandle runtime_;
+  JsContextRef ctx_;
+  std::string desc_;
+
+  static JsValueRef CALLBACK HostFunctionCall(JsValueRef callee, bool isConstructCall, JsValueRef *argumentsIncThis, unsigned short argumentCountIncThis, void *callbackState);
+
+  // String helpers
+  static std::wstring JSStringToSTLWString(JsValueRef str);
+  static std::string JSStringToSTLString(JsValueRef str);
+
+  static JsValueRef CreateJSString(const char*data, size_t length);
+  static JsValueRef CreateJSPropertyId(const char*data, size_t length);
+
+  friend class ChakraJsiRuntimeWithDebugger;

```

Comment:
>ChakraJsiRuntimeWithDebugger [](start = 15, length = 28)

Instead of having ChakraJsiRuntimeWithDebugger be a friend of ChakraJsiRuntime, we should make the members ChakraJsiRuntimeWithDebugger need to access protected.

Reply:
Done.

Reply:
Later I had to revert his .. I'll take care of this in a future change.